spec/integration/inventory_spec.rb

```diff
@@ -3,569 +3,611 @@
 require 'securerandom'
 
 module Hawkular::Inventory::RSpec
-  ENTRYPOINT = 'http://localhost:8080/hawkular/inventory'
+  NON_SECURE_ENTRYPOINT = 'http://localhost:8080/hawkular/inventory'
+  SECURE_ENTRYPOINT = 'https://127.0.0.1:8443/hawkular/inventory'
   DEFAULT_VERSION = '0.17.2.Final'
   VERSION = ENV['INVENTORY_VERSION'] || DEFAULT_VERSION
 
-  include Hawkular::Inventory
-  describe 'Inventory/Tenants', vcr: { decode_compressed_response: true, record: :new_episodes } do
-    it 'Should Get Tenant For Explicit Credentials' do
-      # get the client for given endpoint for given credentials
-      creds = { username: 'jdoe', password: 'password' }
-      mock_inventory_client(VERSION) unless ENV['VCR_UPDATE'] == '1'
-      options = { tenant: 'hawkular' }
-      client = Hawkular::Inventory::InventoryClient.create(entrypoint: ENTRYPOINT,
-                                                           credentials: creds,
-                                                           options: options)
-      tenant = client.get_tenant(creds)
-      expect(tenant).to eq('hawkular')
-    end
+  URL_RESOURCE = 'http://bsd.de'
 
-    it 'Should Get Tenant For Implicit Credentials' do
-      creds = { username: 'jdoe', password: 'password' }
-      mock_inventory_client(VERSION) unless ENV['VCR_UPDATE'] == '1'
-      options = { tenant: 'hawkular' }
-      client = Hawkular::Inventory::InventoryClient.create(entrypoint: ENTRYPOINT,
-                                                           credentials: creds,
-                                                           options: options)
-      tenant = client.get_tenant(creds)
-      expect(tenant).to eq('hawkular')
-    end
-  end
+  NON_SECURE_CONTEXT = 'NonSecure'
+  SECURE_CONTEXT = 'Secure'
 
-  describe 'Inventory Connection' do
-    it 'Should err on bad credentials' do
-      VCR.use_cassette('Inventory/Connection/Should err on bad credentials') do
-        @creds = {
-          username: '-XX-X-jdoe-X',
-          password: 'password'
-        }
-        expect do
-          Hawkular::Inventory::InventoryClient.create(entrypoint: ENTRYPOINT, credentials: @creds)
-        end.to raise_error(Hawkular::BaseClient::HawkularException, 'Unauthorized')
-      end
-    end
-  end
+  [NON_SECURE_CONTEXT, SECURE_CONTEXT].each do |secure_context|
```

Comment:
Souldn't the variable rather be `context` and not `secure_context`, as it can be both secure and non-secure?